packages/api/src/core/prepareParams.ts

```diff
@@ -192,45 +185,42 @@ export default async function prepareParams(operation: Operation, body?: unknown
-        // If no parameters were able to be digested it's because this operation has none so then
-        // we just have to assume that what the user supplied was for a body and not metadata. This
-        // might lead to unwanted false positives if the API definition isn't accurate but short of
-        // throwing an error (one that the user would have no control over resolving anyways) there
-        // isn't anything we can do about it.
-        params.body = merge(params.body, body);
-      } else {
-        const headerParams = caseless({});
-        Object.entries(digestedParameters).forEach(([paramName, param]) => {
-          // Headers are sent case-insensitive so we need to make sure that we're properly
-          // matching them when detecting what our incoming payload looks like.
-          if (param.in === 'header') {
-            headerParams.set(paramName, '');
-          }
-        });
+      const headerParams = caseless({});
+      Object.entries(digestedParameters).forEach(([paramName, param]) => {
+        // Headers are sent case-insensitive so we need to make sure that we're properly
+        // matching them when detecting what our incoming payload looks like.
+        if (param.in === 'header') {
+          headerParams.set(paramName, '');
+        }
+      });
 
-        const intersection = Object.keys(body).filter(value => {
-          if (Object.keys(digestedParameters).includes(value)) {
-            return true;
-          } else if (headerParams.has(value)) {
-            return true;
-          }
-
-          return false;
-        }).length;
-
-        if (intersection && intersection / Object.keys(body).length > 0.25) {
-          /* eslint-disable no-param-reassign */
-          // If more than 25% of the body intersects with the parameters that we've got on hand,
-          // then we should treat it as a metadata object and organize into parameters.
-          metadataIntersected = true;
-          metadata = merge(params.body, body) as Record<string, unknown>;
-          body = undefined;
-          /* eslint-enable no-param-reassign */
-        } else {
-          // For all other cases, we should just treat the supplied body as a body.
-          params.body = merge(params.body, body);
+      // `Accept` headers can't be defined as normal parameters but we should always allow the
+      // user to supply them.
+      if (!headerParams.has('accept')) {
+        headerParams.set('accept', '');
+      }
+
+      const intersection = Object.keys(body).filter(value => {
+        if (Object.keys(digestedParameters).includes(value)) {
+          return true;
+        } else if (headerParams.has(value)) {
+          return true;
         }
+
+        return false;
+      }).length;
+
+      if (intersection && intersection / Object.keys(body).length > 0.25) {
+        /* eslint-disable no-param-reassign */
+        // If more than 25% of the body intersects with the parameters that we've got on hand,
```

Comment:
I have no idea what this means...

Reply:
```js
const body = {
  // post payload
};

const metadata = {
 // query params
 // headers
};

sdk.getSomeOperation(body, metadata);
sdk.getSomeOperation(metadata);
sdk.getSomeOperation(body);
```

Because all three of these operation calls are valid we don't have any way to know if the first item in the call is a body payload or parameters so we do this analysis and intersection work of the first param against what parameters we know about in the spec. If more than 25% of the first parameter intersects with the parameters we know then we assume it's parameters.